crates/libs/core/src/ref.rs

```diff
@@ -0,0 +1,17 @@
+use super::*;
+
+/// A borrowed type with the same memory layout as the type itself that can be used to construct ABI-compatible function signatures.
+#[repr(transparent)]
+pub struct Ref<'a, T: Type<T>>(T::Abi, std::marker::PhantomData<&'a T>);
+
+impl<'a, T: Type<T>> Ref<'a, T> {
+    /// Reads the borrowed value.
+    pub fn read(&self) -> &T::Default {
```

Comment:
Would it make sense for `read` to just be converted to the `Deref` impl?  For COM interfaces, that gives you the ability to call methods directly on it, and you get nice automatic conversions of `&Ref<IFoo>` to `&IFoo`.

Reply:
done!